File: src/main/java/me/austin/event/type/Cancellable.java
package me.austin.event.type;

/**
 * Default implementation of {@link ICancellable}
 * 
 * @author Austin
 */
public abstract class Cancellable implements ICancellable {

    /**
     * This stores whether the event is cancelled or not
     */
    private boolean cancelled = false;

    @Override
    public final boolean isCancelled() {
        return this.cancelled;
    }

    @Override
    public final void setCancelled(boolean cancelled) {
        this.cancelled = cancelled;
    }
}

File: eventbus/src/main/kotlin/me/austin/rush/Cancellable.kt
package me.austin.rush

/**
 * Framework for a cancellable event.
 *
 * @author Austin
 * @since 2022
 */
abstract class Cancellable {
    /**
     * If the event has been cancelled or not.
     *
     * If an event is cancelled then it will not be posted to listeners after it has been cancelled.
     */
    var isCancelled = false
        private set

    /**
     * Use this function to set isCancelled to true.
     */
    fun cancel() {
        this.isCancelled = true
    }
}